src/dune_rules/coq/coq_rules.ml

```diff
@@ -289,7 +293,8 @@ let coqc_file_flags ~dir ~theories_deps ~wrapper_name ~boot_type ~ml_flags
     ; A wrapper_name
     ]
   in
-  [ Bootstrap.flags boot_type; S file_flags ]
+  let boot_flags = Resolve.Memo.read boot_type |> Action_builder.map ~f:Bootstrap.flags in
+  [ Dyn boot_flags; S file_flags ]
```

Comment:
And here we resurface the error, but this time only when we try to execute the relevant rule that needs this.